OnPod/Node/Examples/Solenoids/solenoid.cpp

```diff
@@ -0,0 +1,48 @@
+#include <Arduino.h>
+#include "solenoid.h"
+
+DRV8806::DRV8806(uint8_t PIN_LATCH, uint8_t PIN_DOUT, uint8_t PIN_DIN, uint8_t PIN_SCLK) :
+    PIN_LATCH(PIN_LATCH), PIN_DOUT(PIN_DOUT),
+    PIN_DIN(PIN_DIN), PIN_SCLK(PIN_SCLK)
+    {}
+
+void DRV8806::init() {
+    pinMode(PIN_SCLK, OUTPUT);
+    pinMode(PIN_LATCH, OUTPUT);
+    pinMode(PIN_DOUT, INPUT); // DRV8806 => SAMD21
+    pinMode(PIN_DIN, OUTPUT); // SAMD21 => DRV8806
+    digitalWrite(PIN_SCLK, LOW);
+    digitalWrite(PIN_LATCH, HIGH);
+    digitalWrite(PIN_DOUT, LOW);
+}
+
+void DRV8806::updateSolenoids() {
+    digitalWrite(PIN_LATCH, LOW);
+    digitalWrite(PIN_SCLK, LOW);
+    delayMicroseconds(10);
+
+    // loop through iActiveSolenoids and write DIN to the value of each bit
+    // then write SCLK high then low to shift the bit in
+    for (uint8_t i = 0; i < 8; i++) {
+        uint8_t iSolenoid = 7 - i;
+        bool bState = (iActiveSolenoids >> iSolenoid) & 1;
```

Comment:
try this, same behavior but eliminates the arbitrary i variable, extra - operation and uses u instead of i for unsigned int, i for signed int if going with the hungarian prefixes.

`for (uint8_t uSolenoid = 7; uSolenoid >= 0; uSolenoid--) {
    bool bState = (uActiveSolenoids >> uSolenoid ) & 1;`